app/assets/javascripts/discourse/tests/integration/components/admin-plugins-list-item-test.js

```diff
@@ -0,0 +1,57 @@
+import { render } from "@ember/test-helpers";
+import { hbs } from "ember-cli-htmlbars";
+import { module, test } from "qunit";
+import { setupRenderingTest } from "discourse/tests/helpers/component-test";
+import AdminPlugin from "admin/models/admin-plugin";
+
+module("Integration | Component | admin-plugins-list-item", function (hooks) {
+  setupRenderingTest(hooks);
+
+  function pluginAttrs() {
+    return {
+      id: "discourse-test-plugin",
+      name: "discourse-test-plugin",
+      admin_route: {
+        location: "discourse-test-plugin",
+        label: "admin.plugins.title",
+        use_new_show_route: false,
+        full_location: "admin",
+      },
+      has_settings: true,
+    };
+  }
+
+  test("settings link route", async function (assert) {
+    this.currentUser.admin = true;
+    this.plugin = AdminPlugin.create(pluginAttrs());
```

Comment:
It's preferred to use the store to create new model instances:

```suggestion
    const store = getOwner(this).lookup("service:store");
    this.plugin = store.createRecord("admin-plugin", pluginAttrs());
```

(currently those are equivalent but the `.create()` requires ugly discourse-specific hacks under the hood which we'd like to remove in the future)

Reply:
@CvX I'm not following sorry, how does looking up service:store work here? And in this case `AdminPlugin.create` is a normal static class helper function that calls the constructor like this:

https://github.com/discourse/discourse/blob/3555ca0965e4b50398fd0d52544a2a809e0dfd1e/app/assets/javascripts/admin/addon/models/admin-plugin.js#L5-L30

Reply:
Looking up the service is done just like that. Import the function:

```js
import { getOwner } from "@ember/owner";
```

and then in tests use it to look up the store:

```js
const store = getOwner(this).lookup("service:store");
```

`this` in integration/acceptance tests is an object which owner is an application instance.
And the store looks up the class based on the model name you provide:

```js
this.plugin = store.createRecord("admin-plugin", { commit_url: "test" });
this.plugin.commitUrl === "test";
```